pydicer/analyse/data.py

```diff
@@ -0,0 +1,261 @@
+import logging
+from pathlib import Path
+import json
+import SimpleITK as sitk
+import pandas as pd
+import pydicom
+
+from radiomics import firstorder, shape, glcm, glrlm, glszm, ngtdm, gldm, imageoperations
+
+logger = logging.getLogger(__name__)
+
+PYRAD_DEFAULT_SETTINGS = {
+    "binWidth": 25,
+    "resampledPixelSpacing": None,
+    "interpolator": "sitkNearestNeighbor",
+    "verbose": True,
+    "removeOutliers": 10000,
+}
+
+AVAILABLE_RADIOMICS = {
+    "firstorder": firstorder.RadiomicsFirstOrder,
+    "shape": shape.RadiomicsShape,
+    "glcm": glcm.RadiomicsGLCM,
+    "glrlm": glrlm.RadiomicsGLRLM,
+    "glszm": glszm.RadiomicsGLSZM,
+    "ngtdm": ngtdm.RadiomicsNGTDM,
+    "gldm": gldm.RadiomicsGLDM,
+}
+
+FIRST_ORDER_FEATURES = firstorder.RadiomicsFirstOrder.getFeatureNames()
+DEFAULT_RADIOMICS = {
+    "firstorder": [f for f in FIRST_ORDER_FEATURES if not FIRST_ORDER_FEATURES[f]]
+}
+
+
+class AnalyseData:
+    """
+    Class that performs common analysis on converted data
+
+    Args:
+        data_directory (str|pathlib.Path, optional): Directory in which data is stored. Defaults to
+            ".".
+        dataset (str, optional): The name of the dataset on which to run analysis. Defaults to
+            "nifti".
+    """
+
+    def __init__(self, data_directory=".", dataset_name="nifti"):
+        self.data_directory = Path(data_directory)
+        self.dataset_name = dataset_name
+        self.dataset_directory = self.data_directory.joinpath(dataset_name)
+
+    def get_all_computed_radiomics_for_dataset(self):
+        """Return a DataFrame of radiomics computed for this dataset
+
+        Returns:
+            pd.DataFrame: The DataFrame of all radiomics computed for dataset
+        """
+
+        dfs = []
+        for radiomics_file in self.dataset_directory.glob("**/structures/*.csv"):
+            dfs.append(pd.read_csv(radiomics_file, index_col=0))
+
+        df = pd.concat(dfs)
+        df.sort_values(["Patient", "Contour"], inplace=True)
```

Comment:
Is this sort order conventional and could user input on potential ordering be useful?

Reply:
I'd imagine that usually you would want the output grouped by patient first and then by the contour... And this function returns a dataframe so the user can just sort that themselves if they want something different?